### PR TITLE
Remove redundant nil checks around a range loop

### PR DESCRIPTION
**Pull request by teresy**
**What this PR does / why we need it**:

Simplifies code by removing redundant nil checks for loops over slices or maps.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
@teresy would you sign `cla/linuxfoundation` first?

File: pkg/apis/autoscaling/v1/conversion.go [hunk 1]
```diff
@@ -328,10 +328,8 @@ func Convert_autoscaling_HorizontalPodAutoscaler_To_v1_HorizontalPodAutoscaler(i
 	if len(otherMetrics) > 0 || len(in.Status.CurrentMetrics) > 0 || len(currentConditions) > 0 {
 		old := out.Annotations
 		out.Annotations = make(map[string]string, len(old)+3)
-		if old != nil {
-			for k, v := range old {
-				out.Annotations[k] = v
-			}
+		for k, v := range old {
```
[the remark this answers is not in the document]
Comment (teresy) [reply pkg/apis/autoscaling/v1/conversion.go]: Good eye, thanks for catching.

**Comment by teresy** [comment]
Friendly ping @andyzhangx

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request closed without merging**